external/solr/src/main/java/com/digitalpebble/storm/crawler/solr/persistence/SolrSpout.java

```diff
@@ -172,20 +172,28 @@ private void populateBuffer() {
             QueryResponse response = connection.getClient().query(query);
             SolrDocumentList docs = new SolrDocumentList();
 
+            int numhits = response.getResults().size();
+      
             if (StringUtils.isNotBlank(diversityField)) {
-                Map<String, SolrDocumentList> expandedResults = response
-                        .getExpandedResults();
-
-                for (String key : expandedResults.keySet()) {
-                    docs.addAll(expandedResults.get(key));
+              //get expand result
+              Map<String, SolrDocumentList> expandedResults = response
+                      .getExpandedResults();
+
+              for (SolrDocument doc : response.getResults()) {
+                String df = (String) doc.get(diversityField);
```

Comment:
The main caveat I see here is that we're iterating twice on the main result set, since the CollapsingQParser already "collapsed" the main result set into one document per value in the bucket field (i.e. the host field) all we need to do is "merge" the results in the main result set with the ones already extracted from the expanded section. 

Basically something like:
```java
docs.addAll(response.getResults());
```
This guarantees that the `docs` var contains all the expanded results and also the "main" document in the results list, without the caveat of iterating over the main result set twice.